elastic/kibana
# [ILM] Delete index_codec option if disabled in UI

Fixes https://github.com/elastic/kibana/issues/87219

I could not reproduce on `7.10`, so I think this was introduced at some point during the `7.11` release cycle, possibly from https://github.com/elastic/kibana/pull/83077.

**How to test:**
1. Navigate to ILM and create a policy.
2. Enable the force merge action for the hot phase. Specify the number of segments and enable the "Compress stored fields" option.
3. Enable the force merge action for the warm phase. Specify the number of segments and enable the "Compress stored fields" option.
4. Save the policy.
5. Reopen the policy and disable the "Compress stored fields" option for both hot and warm phases.
6. Save the policy.
7. Reopen the policy and verify the "Compress stored fields" options are disabled.
